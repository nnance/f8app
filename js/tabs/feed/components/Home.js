import React from 'react';
import {
  View,
  Image,
  TouchableOpacity,
} from 'react-native';
import Navbar from './NavBar';
import FeedList from './FeedList';

class Home extends React.Component {
  render() {
    return (
      <View style={{ flex: 1, backgroundColor: '#FFFFFF' }}>
        <Navbar
          renderRightMenu={() => (
            <TouchableOpacity><Image style={{ height: 20, resizeMode: 'contain' }} source={require('../img/search.png')} /></TouchableOpacity>
          )}
          renderTitle={() => (
            <Image style={{ width: 80, height: 30, resizeMode: 'contain' }} source={require('../img/title.png')} />
          )}
          titleStyle={{
            flex: 1,
            alignItems: 'flex-start',
          }}
        />
        <FeedList />
      </View>
    );
  }
}

export default Home;
