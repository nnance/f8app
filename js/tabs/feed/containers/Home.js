import gql from 'graphql-tag';
import { graphql } from 'react-apollo';

import FeedHome from '../components/FeedHome';

export const query = gql`
{
  feed: feedEditor(filter: {userId: "58b62c5fda82d62bae708b3b"}) {
    author {
      name
      profilePicture
    }
    clog {
      createdAt
      clog {
        id
        title
        coverImage
        category
        synopsis
      }
    }
  }
}
`;

const mapQueryToProps = ({ data }) => {
  const { loading, feed } = data;
  return ({
    feed: loading ? [] : feed,
    loading,
  });
};

export default graphql(query, {
  props: mapQueryToProps,
})(FeedHome);
