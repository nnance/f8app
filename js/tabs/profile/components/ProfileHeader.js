import React from 'react';
import {
  View,
  StyleSheet,
} from 'react-native';

import ProfilePicture from '../../../common/ProfilePicture';
import ProfileCover from './ProfileCover';

const styles = StyleSheet.create({
  header: {
    justifyContent: 'center',
    alignItems: 'center',
    height: undefined,
    width: undefined,
    paddingVertical: 10,
  },
  headerBackground: {
    justifyContent: 'center',
    alignItems: 'center',
    height: undefined,
    width: undefined,
  },
});

class ProfileHeader extends React.Component {
  render() {
    return (
      <ProfileCover
        customSource={this.props.customCoverSource}
        user={this.props.user}
        style={styles.headerBackground}
      >
        <View style={styles.header}>
          <ProfilePicture
            size={100}
            user={this.props.user}
            customSource={this.props.customSource}
          />
          {this.props.children}
        </View>
      </ProfileCover>
    );
  }
}

export default ProfileHeader;
