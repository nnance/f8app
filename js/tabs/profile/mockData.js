export const user = {
  id: 1,
  name: 'test',
  sex: 'M',
  facebookLinked: false,
};
export const followingCount = 1800;
export const followerCount = 1800;
export const following = [{ name: 'Art Nattapat' }, { name: 'Art Art Art' }, { name: 'Art Art Art' }, { name: 'Art Art Art' }, { name: 'Art Art Art' }, { name: 'Art Art Art' }];
export const follower = [{ name: 'Art Nattapat', following: true }, { name: 'Art Art Art', following: false }, { name: 'Art Art Art', following: false }, { name: 'Art Art Art', following: true }, { name: 'Art Art Art', following: true }, { name: 'Art Art Art', following: false }];
export const myFan = [{ name: 'Art Nattapat', candys: 1250 }, { name: 'Art Art Art', candys: 1250 }, { name: 'Art Art Art', candys: 1000 }, { name: 'Art Art Art', candys: 789 }, { name: 'Art Art Art', candys: 7013 }, { name: 'Art Art Art', candys: 0 }];
export const candys = 1890;
export const activity = [{
  activity: 'like',
  title: 'แมค เดมอน ตะลุยอวกาศ',
  outline: 'The Martain กำกับโดย ริดลีย์ สก๊อต เนื้อหาเล่าถึง นักบินอวกาศที่ถูกทิ้งไว้บนดาวอังคาร',
  uri: 'http://3.bp.blogspot.com/-zW6wqY_1Me0/VYvJMOV4mcI/AAAAAAAAD-4/mB_AxhFoJH4/s1600/178491main_sig07-009-516.jpg',
  date: new Date(2016, 10, 4),
}, {
  activity: 'like',
  title: 'แมค เดมอน ตะลุยอวกาศ',
  outline: 'The Martain กำกับโดย ริดลีย์ สก๊อต เนื้อหาเล่าถึง นักบินอวกาศที่ถูกทิ้งไว้บนดาวอังคาร',
  uri: 'http://3.bp.blogspot.com/-zW6wqY_1Me0/VYvJMOV4mcI/AAAAAAAAD-4/mB_AxhFoJH4/s1600/178491main_sig07-009-516.jpg',
  date: new Date(2016, 10, 8),
}, {
  activity: 'read',
  title: 'แมค เดมอน ตะลุยอวกาศ',
  outline: 'The Martain กำกับโดย ริดลีย์ สก๊อต เนื้อหาเล่าถึง นักบินอวกาศที่ถูกทิ้งไว้บนดาวอังคาร',
  uri: 'http://www.fujisan.ne.jp/fit/th/images/ohishikouen430.jpg',
  date: new Date(2015, 10, 8),
}];

export const myClogs = [
  {
    title: 'Richy Rich! รวยมากนะ! รู้ยังคะทุกคน',
    author: {
      name: 'David Beckham',
    },
    preview: require('../../assets/A.png'),
    category: 'M',
    viewCount: 12300,
    likeCount: 1500,
    createdAt: new Date(2015, 5, 24),
  },
  {
    title: 'Money Honey คุณชายหน้าตายกับยัยขี้งก',
    author: {
      name: 'สุดสาครนอย ไทรโยค',
    },
    preview: require('../../assets/B.png'),
    category: 'N',
    viewCount: 100,
    likeCount: 55,
    createdAt: new Date(2016, 10, 8),
  },
];

export const bookmark = [
  {
    title: 'Richy Rich! รวยมากนะ! รู้ยังคะทุกคน',
    preview: require('../../assets/A.png'),
    category: 'N',
    bookmarkCount: 7,
  },
  {
    title: 'Money Honey คุณชายหน้าตายกับยัยขี้งก',
    preview: require('../../assets/B.png'),
    category: 'M',
    bookmarkCount: 3,
  },
];
