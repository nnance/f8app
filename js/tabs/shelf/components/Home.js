import React from 'react';

import {
  View,
  Image,
  TouchableOpacity,
  StyleSheet,
} from 'react-native';
import gql from 'graphql-tag';

import NavBar from './NavBar';
import BorderButton from '../../../common/BorderButton';
import FixBugScrollView from '../../../common/FixBugScrollView';

import MetaClogListView from './MetaClogListView';
import RecommendedClog from './RecommendedClog';
import ExploreCategory from './ExploreCategory';
import HeroBanner from './HeroBanner';

const styles = StyleSheet.create({
  clogListButton: {
    flex: 1,
    width: 80,
    paddingHorizontal: 0,
    alignItems: 'center',
  },
});

class Home extends React.Component {
  constructor(...args) {
    super(...args);

    this.renderFollowButton = this.renderFollowButton.bind(this);
    this.renderUnfollowButton = this.renderUnfollowButton.bind(this);
    this.renderTrendingButton = this.renderTrendingButton.bind(this);
  }

  renderFollowButton() {
    // remove when implement
    /* eslint class-methods-use-this: off */
    return (
      <BorderButton caption="ติดตาม" type="fadedWhite" containerStyle={styles.clogListButton} />
    );
  }

  renderUnfollowButton() {
    // remove when implement
    /* eslint class-methods-use-this: off */
    return (
      <BorderButton caption="เลิกติดตาม" type="fadedGrey" containerStyle={styles.clogListButton} />
    );
  }

  renderTrendingButton() {
    return (
      <BorderButton
        caption="ทั้งหมด" type="fadedWhite" containerStyle={styles.clogListButton}
        onPress={() => this.props.goToTrendingListView({ title: 'ยอดนิยม' })}
      />
    );
  }

  render() {
    return (
      <View style={{ flex: 1, backgroundColor: 'rgb(81, 20, 64)' }}>
        <NavBar
          renderLeftMenu={() => (
            <TouchableOpacity onPress={this.props.onOpenShelfMenu}>
              <View style={{ height: 40, flex: 1, justifyContent: 'center', alignItems: 'flex-start' }}>
                <Image style={{ height: 15, resizeMode: 'contain' }} source={require('../../../assets/shelf/menu.png')} />
              </View>
              </TouchableOpacity>
          )}
          renderRightMenu={() => (
            <TouchableOpacity><Image style={{ height: 20, resizeMode: 'contain' }} source={require('../../../assets/common/pink-search.png')} /></TouchableOpacity>
          )}
          renderTitle={() => (
            <View style={{ height: 30, flex: 1, justifyContent: 'center', alignItems: 'center' }}>
              <Image style={{ height: 30, resizeMode: 'contain' }} source={require('../../../assets/shelf/title.png')} />
            </View>
          )}
          titleStyle={{
            flex: 1,
            alignItems: 'flex-start',
          }}
        />
        <FixBugScrollView>
          {/*<Image source={require('../../../assets/shelf/home-bg-1.png')} style={{ width: undefined, height: 700, resizeMode: 'stretch', backgroundColor: 'transparent' }}>
            <View style={{ flex: 2 }}>
              <HeroBanner goToBook={this.props.goToBook} clogs={this.props.heroBanners} />
            </View>
            <View style={{ flex: 1, paddingLeft: 10 }}>
              <MetaClogListView goToBook={this.props.goToBook} header="TRENDING" clogs={this.props.trendingClogs} renderButton={this.renderTrendingButton} />
            </View>
            <View style={{ flex: 1.6 }}>
              <RecommendedClog clog={this.props.recommendedClog} goToBook={this.props.goToBook} />
            </View>
          </Image>
          */}
          <Image source={require('../../../assets/shelf/home-bg-1.5.png')} style={{ resizeMode: 'stretch', backgroundColor: 'transparent', width: undefined, height: undefined, paddingTop: 20 }}>
            {
              this.props.favoriteTags.map(tag => (
                <View key={tag.id} style={{ paddingLeft: 0, paddingTop: 10 }}>
                  <MetaClogListView
                    goToBook={this.props.goToBook}
                    header={tag.name.toUpperCase()}
                    clogs={tag.trendingClogs}
                    renderButton={
                      !tag.following ?
                          this.renderFollowButton
                        : this.renderUnfollowButton
                    }
                  />
                </View>
              ))
            }
          </Image>
          <ExploreCategory onPress={category => this.props.goToClogCategory(category)} />
        </FixBugScrollView>
      </View>
    );
  }
}

Home.fragments = {
  HeroBanner: HeroBanner.fragments.clog,
  RecommendedClog: RecommendedClog.fragments.clog,
  MetaClogListView: MetaClogListView.fragments.clog,
  FavoritTag: gql`
    fragment FavoritTag on Tag {
      id
      name
      trendingClogs(limit: 10) {
        ...MetaClogListView
      }
    }
  `,
};

export default Home;
