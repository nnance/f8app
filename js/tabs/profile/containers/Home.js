import { connect } from 'react-redux';

import { logOutWithPrompt } from '../../../actions';

import Home from '../components/Home';

export default connect(
    state => ({
      user: state.user,
      followingCount: 1800,
      followerCount: 1800,
      candys: 100,
    }),
  {
    logOutWithPrompt,
  },
)(Home);
