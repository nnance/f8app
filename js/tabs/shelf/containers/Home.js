import gql from 'graphql-tag';
import { graphql } from 'react-apollo';

import Home from '../components/Home';
import { CLOG_PREVIEW_LIMIT } from '../constants';

export const query = gql`
  query {
    trendingClogs(limit: ${CLOG_PREVIEW_LIMIT}) {
      ...MetaClogListView
    }
    recommendedClog {
      ...RecommendedClog
    }
    favoriteTags: tags(limit: 3) {
      ...FavoritTag
    }
    heroBanners {
      ...HeroBanner
    }
  }
  ${Home.fragments.RecommendedClog}
  ${Home.fragments.HeroBanner}
  ${Home.fragments.MetaClogListView}
  ${Home.fragments.FavoritTag}
`;

export const mapQueryToProps = ({ data }) => {
  const { loading, trendingClogs, recommendedClog, favoriteTags, heroBanners, error } = data;
  if (error) {
    console.error('graphql error: ', error);
  }
  return ({
    trendingClogs: loading || !!error ? [] : trendingClogs,
    recommendedClog,
    heroBanners: loading || !!error ? [] : heroBanners,
    favoriteTags: loading || !!error ? [] : favoriteTags,
  });
};

export default graphql(query, {
  props: mapQueryToProps,
})(Home);
