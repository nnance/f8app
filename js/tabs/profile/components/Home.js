import React from 'react';
import {
  Image,
  Text,
  View,
  StyleSheet,
  TouchableOpacity,
} from 'react-native';
import PureListView from '../../../common/PureListView';
import FixBugScrollView from '../../../common/FixBugScrollView';
import { toHumanNumber } from '../../../common/utils';

import ProfileHeader from './ProfileHeader';

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  menuList: {
    flex: 1,
    backgroundColor: 'white',
  },
  row: {
    padding: 15,
    flex: 1,
    flexDirection: 'row',
  },
  menuIcon: {
    width: 20,
    height: 20,
    resizeMode: 'contain',
  },
  menuText: {
    paddingLeft: 20,
  },
  nameContainer: {
    width: 200,
    padding: 10,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
  },
  name: {
    color: 'white',
    fontWeight: 'bold',
    fontSize: 20,
  },
  smallText: {
    color: 'white',
    fontWeight: 'bold',
  },
  headNumberDetail: {
    minWidth: 60,
    paddingTop: 3,
    paddingBottom: 3,
    paddingLeft: 10,
    paddingRight: 10,
    backgroundColor: 'rgba(0, 0, 0, 0.2)',
    borderRadius: 5,
    alignItems: 'center',
  },
  whiteLine: {
    borderRightWidth: 1,
    borderColor: 'rgba(255, 255, 255, 0.5)',
    height: 10,
  },
  numberDetail: {
    alignItems: 'center',
    paddingLeft: 20,
    paddingRight: 20,
    borderColor: 'rgba(255, 255, 255, 0.3)',
  },
});

const menuList = [
  {
    name: 'bookmark',
    icon: require('../../../assets/profile-icon/bookmark.png'),
    title: 'Bookmark',
  },
  {
    name: 'myclog',
    icon: require('../../../assets/profile-icon/myclog.png'),
    title: 'Clog ของฉัน',
  },
  {
    name: 'myfan',
    icon: require('../../../assets/profile-icon/myfan.png'),
    title: 'แฟนคลับของฉัน',
  },
  {
    name: 'jellyShop',
    icon: require('../../../assets/profile-icon/candy-shop.png'),
    title: 'Jelly Shop',
  },
  {
    name: 'activity',
    icon: require('../../../assets/profile-icon/activity.png'),
    title: 'กิจกรรม',
  },
  {
    name: 'logout',
    icon: require('../../../assets/profile-icon/logout.png'),
    title: 'Logout',
  },
];

const NumberDetail = props => (
  <TouchableOpacity
    style={[styles.numberDetail, { borderRightWidth: props.borderRight ? 1 : 0 }]}
    onPress={() => props.onPress && props.onPress()}
  >
    <View style={styles.headNumberDetail}>
      <Text style={styles.smallText}>
        {toHumanNumber(props.number)}
      </Text>
    </View>
    <Text style={styles.smallText}>
      {props.title}
    </Text>
  </TouchableOpacity>
  );

const CandyCorner = props => (
  <View
    style={{
      position: 'absolute',
      left: 0,
      top: 20,
      flex: 1,
      flexDirection: 'row',
      justifyContent: 'center',
      alignItems: 'center',
    }}
  >
    <Image
      source={require('../../../assets/profile-icon/candy.png')}
      style={{
        width: 15,
        height: 15,
        marginLeft: 5,
        marginRight: 5,
      }}
    />
    <View
      style={{
        paddingTop: 2,
        paddingBottom: 2,
        paddingLeft: 5,
        paddingRight: 5,
        backgroundColor: 'rgba(255, 255, 255, 0.1)',
        borderRadius: 5,
        borderWidth: 1,
        borderColor: 'white',
      }}
    >
      <Text style={{ color: 'white', fontSize: 13 }}>{toHumanNumber(props.candys)}</Text>
    </View>
  </View>
);

class Home extends React.Component {
  constructor(...args) {
    super(...args);
    this.renderMenu = this.renderMenu.bind(this);
    this.onLogout = this.onLogout.bind(this);
  }

  onLogout() {
    this.props.logOutWithPrompt();
  }

  renderMenu(menu) {
    return (
      <TouchableOpacity
        title={menu.name}
        onPress={() => {
          if (menu.name === 'logout') {
            this.onLogout();
          } else if (this.props.onMenuPress) {
            this.props.onMenuPress(menu.name);
          }
        }}
      >
        <View style={styles.row}>
          <Image style={styles.menuIcon} source={menu.icon} />
          <Text style={styles.menuText}>{menu.title}</Text>
        </View>
      </TouchableOpacity>
    );
  }

  render() {
    const name = this.props.user.name || '';
    return (
      <View style={styles.container}>
        <ProfileHeader user={this.props.user}>
          <View style={styles.nameContainer}>
            <Text numberOfLines={1} style={styles.name}>
              {name.toUpperCase()}
            </Text>
            <TouchableOpacity onPress={this.props.onEditProfile}>
              <Image
                style={{
                  marginLeft: 10,
                  height: 20,
                  width: 20,
                  backgroundColor: 'transparent',
                }}
                source={require('../../../assets/profile-icon/edit-profile.png')}
              />
            </TouchableOpacity>
          </View>
          <View style={{ flexDirection: 'row', alignItems: 'center', justifyContent: 'center' }}>
            <NumberDetail title="ผู้ติดตาม" number={this.props.followerCount} borderRight onPress={() => this.props.onFollowerPress && this.props.onFollowerPress()} />
            <NumberDetail title="กำลังติดตาม" number={this.props.followingCount} borderRight onPress={() => this.props.onFollowingPress && this.props.onFollowingPress()} />
            <NumberDetail title="Candys" number={this.props.candys} onPress={() => this.props.onCandyPress && this.props.onCandyPress()} />
          </View>
        </ProfileHeader>
        <View style={styles.menuList}>
          <FixBugScrollView>
            <PureListView
              minContentHeight={0}
              title="Profile"
              data={menuList}
              renderRow={this.renderMenu}
            />
          </FixBugScrollView>
        </View>
        <CandyCorner candys={this.props.candys} />
      </View>
    );
  }
}

export default Home;
