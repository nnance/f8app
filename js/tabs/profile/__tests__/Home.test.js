import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { View, TouchableOpacity } from 'react-native';

import ProfileHeader from '../components/ProfileHeader';
import Home from '../components/Home';
import * as mockData from '../mockData';

describe('Home', () => {
  let wrapper;
  const followingSpy = jest.fn();
  const followerSpy = jest.fn();
  const candySpy = jest.fn();
  const editProfileSpy = jest.fn();
  const menuPressSpy = jest.fn();
  beforeEach(() => {
    wrapper = shallow(<Home
      {...mockData}
      onFollowerPress={followingSpy}
      onFollowingPress={followerSpy}
      onCandyPress={candySpy}
      onEditProfile={editProfileSpy}
      onMenuPress={menuPressSpy}
    />);
  });

  afterEach(() => {
    followingSpy.mockClear();
    followerSpy.mockClear();
    candySpy.mockClear();
    editProfileSpy.mockClear();
    menuPressSpy.mockClear();
  });

  it('render', () => {
    const tree = renderer.create(<Home {...mockData} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('press following', () => {
    wrapper.find('[title="ผู้ติดตาม"]').simulate('press');
    expect(followingSpy).toBeCalled();
  });

  it('press following', () => {
    wrapper.find('[title="กำลังติดตาม"]').simulate('press');
    expect(followerSpy).toBeCalled();
  });

  it('press following', () => {
    wrapper.find('[title="Candys"]').simulate('press');
    expect(candySpy).toBeCalled();
  });

  it('press editProfile', () => {
    wrapper.find(ProfileHeader).find(TouchableOpacity).simulate('press');
    expect(editProfileSpy).toBeCalled();
  });

  it('renderMenu', () => {
    const component = wrapper.instance().renderMenu({ name: 'following' });
    const tree = renderer.create(component);
    expect(tree.toJSON()).toMatchSnapshot();
    const menuWrapper = shallow(<View>{component}</View>);
    menuWrapper.find('[title="following"]').simulate('press');
    expect(menuPressSpy).toBeCalledWith('following');
  });
});
