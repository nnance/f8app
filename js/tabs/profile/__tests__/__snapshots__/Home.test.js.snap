exports[`Home render 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }>
  <Image
    customSource={undefined}
    source={null}
    style={
      Array [
        Object {
          "resizeMode": "cover",
        },
        Object {
          "alignItems": "center",
          "height": undefined,
          "justifyContent": "center",
          "width": undefined,
        },
      ]
    }
    user={
      Object {
        "facebookLinked": false,
        "id": 1,
        "name": "test",
        "sex": "M",
      }
    }>
    <View
      style={
        Object {
          "alignItems": "center",
          "height": undefined,
          "justifyContent": "center",
          "paddingVertical": 10,
          "width": undefined,
        }
      }>
      <Image
        source={
          Object {
            "uri": "http://graph.facebook.com/1/picture?width=200&height=200",
          }
        }
        style={
          Object {
            "borderRadius": 50,
            "height": 100,
            "width": 100,
          }
        } />
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 10,
            "width": 200,
          }
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "color": "white",
              "fontSize": 20,
              "fontWeight": "bold",
            }
          }>
          TEST
        </Text>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}>
          <Image
            source={1}
            style={
              Object {
                "backgroundColor": "transparent",
                "height": 20,
                "marginLeft": 10,
                "width": 20,
              }
            } />
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderColor": "rgba(255, 255, 255, 0.3)",
              "borderRightWidth": 1,
              "opacity": 1,
              "paddingLeft": 20,
              "paddingRight": 20,
            }
          }
          testID={undefined}>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(0, 0, 0, 0.2)",
                "borderRadius": 5,
                "minWidth": 60,
                "paddingBottom": 3,
                "paddingLeft": 10,
                "paddingRight": 10,
                "paddingTop": 3,
              }
            }>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "white",
                  "fontWeight": "bold",
                }
              }>
              1,800
            </Text>
          </View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
                "fontWeight": "bold",
              }
            }>
            ผู้ติดตาม
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderColor": "rgba(255, 255, 255, 0.3)",
              "borderRightWidth": 1,
              "opacity": 1,
              "paddingLeft": 20,
              "paddingRight": 20,
            }
          }
          testID={undefined}>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(0, 0, 0, 0.2)",
                "borderRadius": 5,
                "minWidth": 60,
                "paddingBottom": 3,
                "paddingLeft": 10,
                "paddingRight": 10,
                "paddingTop": 3,
              }
            }>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "white",
                  "fontWeight": "bold",
                }
              }>
              1,800
            </Text>
          </View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
                "fontWeight": "bold",
              }
            }>
            กำลังติดตาม
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderColor": "rgba(255, 255, 255, 0.3)",
              "borderRightWidth": 0,
              "opacity": 1,
              "paddingLeft": 20,
              "paddingRight": 20,
            }
          }
          testID={undefined}>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(0, 0, 0, 0.2)",
                "borderRadius": 5,
                "minWidth": 60,
                "paddingBottom": 3,
                "paddingLeft": 10,
                "paddingRight": 10,
                "paddingTop": 3,
              }
            }>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "white",
                  "fontWeight": "bold",
                }
              }>
              1,890
            </Text>
          </View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
                "fontWeight": "bold",
              }
            }>
            Candys
          </Text>
        </View>
      </View>
    </View>
  </Image>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "flex": 1,
      }
    }>
    <ScrollView
      onScroll={[Function]}>
      <ScrollView
        contentInset={
          Object {
            "bottom": 0,
            "top": 0,
          }
        }
        data={
          Array [
            Object {
              "icon": 1,
              "name": "bookmark",
              "title": "Bookmark",
            },
            Object {
              "icon": 1,
              "name": "myclog",
              "title": "Clog ของฉัน",
            },
            Object {
              "icon": 1,
              "name": "myfan",
              "title": "แฟนคลับของฉัน",
            },
            Object {
              "icon": 1,
              "name": "jellyShop",
              "title": "Jelly Shop",
            },
            Object {
              "icon": 1,
              "name": "activity",
              "title": "กิจกรรม",
            },
            Object {
              "icon": 1,
              "name": "logout",
              "title": "Logout",
            },
          ]
        }
        dataSource={
          ListViewDataSource {
            "items": 6,
          }
        }
        initialListSize={1}
        minContentHeight={0}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={1000}
        onKeyboardDidHide={undefined}
        onKeyboardDidShow={undefined}
        onKeyboardWillHide={undefined}
        onKeyboardWillShow={undefined}
        onLayout={[Function]}
        onScroll={[Function]}
        pageSize={1}
        removeClippedSubviews={true}
        renderEmptyList={null}
        renderFooter={[Function]}
        renderRow={[Function]}
        renderSeparator={[Function]}
        scrollEventThrottle={50}
        scrollRenderAheadDistance={1000}
        stickyHeaderIndices={Array []}
        title="Profile">
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}>
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "padding": 15,
              }
            }>
            <Image
              source={1}
              style={
                Object {
                  "height": 20,
                  "resizeMode": "contain",
                  "width": 20,
                }
              } />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "paddingLeft": 20,
                }
              }>
              Bookmark
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#eeeeee",
              "height": 1,
            }
          } />
      </ScrollView>
    </ScrollView>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "top": 20,
      }
    }>
    <Image
      source={1}
      style={
        Object {
          "height": 15,
          "marginLeft": 5,
          "marginRight": 5,
          "width": 15,
        }
      } />
    <View
      style={
        Object {
          "backgroundColor": "rgba(255, 255, 255, 0.1)",
          "borderColor": "white",
          "borderRadius": 5,
          "borderWidth": 1,
          "paddingBottom": 2,
          "paddingLeft": 5,
          "paddingRight": 5,
          "paddingTop": 2,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 13,
          }
        }>
        1,890
      </Text>
    </View>
  </View>
</View>
`;

exports[`Home renderMenu 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "padding": 15,
      }
    }>
    <Image
      source={undefined}
      style={
        Object {
          "height": 20,
          "resizeMode": "contain",
          "width": 20,
        }
      } />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "paddingLeft": 20,
        }
      } />
  </View>
</View>
`;
